python/mypy.ini

```diff
@@ -1,6 +1,6 @@
 [mypy]
 no_implicit_optional = True
-python_version = 3.7
+python_version = 3.12
```

Comment:
Worth bumping the Python version in the CI as well ([here](https://github.com/GaloisInc/argo/blob/59823e8fd6985986abd9ae784932b94bd735b17d/.github/workflows/ci.yml#L22)). Make sure to also leave a comment here pointing to this part of the CI configuration (and vice versa) so that if we bump the Python version in the future, we remember to update both locations.